File: backends/java/zappy-domain/src/main/java/com/zappymart/domain/catalogue/Category.java
package com.zappymart.domain.catalogue;

import java.util.Objects;

public record Category(String id, String name, String slug) {

    public Category {
        Objects.requireNonNull(id, "id");
        Objects.requireNonNull(name, "name");
        Objects.requireNonNull(slug, "slug");
    }
}

File: backends/dotnet/src/Zappy.Domain/Catalogue/Category.cs
namespace Zappy.Domain;

public sealed class Category
{
    private Category()
    {
    }

    public Category(string id, string name, string slug, int catalogueOrder)
    {
        Id = id;
        Name = name;
        Slug = slug;
        CatalogueOrder = catalogueOrder;
    }

    public string Id { get; private set; } = null!;

    public string Name { get; private set; } = null!;

    public string Slug { get; private set; } = null!;

    public int CatalogueOrder { get; private set; }
}
